torch/cuda/comm.py

```diff
@@ -94,6 +94,7 @@ def scatter(tensor, devices, chunk_sizes=None, dim=0):
         assert min(chunk_sizes) > 0, "got a negative chunk_size"
         chunks = [tensor.narrow(dim, start - size, size)
                   for start, size in zip(_accumulate(chunk_sizes), chunk_sizes)]
+    chunks = tuple(chunk.contiguous() for chunk in chunks)
```

Comment:
Worried about transfer performance. Transferring peer to peer of contiguous should be faster so would like to do before.